### PR TITLE
update plugin to work with Bamboo 8+

### PR DESCRIPTION
**Pull request by rain-on**
Updated the bamboo.version from 7.2.6 to 8.0.4 to overcome a recognised vulnerability in the base of Bamboo.

The CVE was announced as 
https://confluence.atlassian.com/security/multiple-products-security-advisory-unrendered-unicode-bidirectional-override-characters-cve-2021-42574-1086419475.html?subid=1529035734&jobid=105251301&utm_campaign=multiple-products-advisory_november-2021_EML-11741&utm_medium=email&utm_source=alert-email

The mocked build logger required update to be 8+ compatible, but otherwise the production code base appears to be already compliant.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
     <properties>
         <maven.compiler.source>1.7</maven.compiler.source>
         <maven.compiler.target>1.7</maven.compiler.target>
-        <bamboo.version>7.2.1</bamboo.version>
+        <bamboo.version>8.0.4</bamboo.version>
         <bamboo.data.version>7.2.1</bamboo.data.version>
```
Comment (long-sean) [inline pom.xml:49]: The project `README` indicates that both `bamboo.version` and `bamboo.data.version` should be increased...?

> Edit the pom.xml file and set <bamboo.version>6.2.1</bamboo.version> and > <bamboo.data.version>6.2.1</bamboo.data.version> (the actual versions will be different to these examples) to the latest Bamboo version. This in turn will mean atlas-run will load the specified Bamboo version.

Comment (rain-on) [reply pom.xml:49]: yeah probably should ... but somehow ... still works!

File: src/test/java/com/octopus/services/impl/RecordingBuildLogger.java [hunk 2]
```diff
@@ -14,6 +14,8 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
```
Comment (long-sean) [inline src/test/java/com/octopus/services/impl/RecordingBuildLogger.java:17]: Unused import

**Review by long-sean — APPROVED** [review]
LGTM

File: src/test/java/com/octopus/services/impl/RecordingBuildLogger.java (hunk 2 above)
Comment (rain-on) [reply src/test/java/com/octopus/services/impl/RecordingBuildLogger.java]: removed

**Pull request merged**